src/Raven.Studio/typescript/viewmodels/resources/widgets/clusterOverviewWidget.ts

```diff
@@ -0,0 +1,288 @@
+import appUrl = require("common/appUrl");
+import router = require("plugins/router");
+import moment = require("moment");
+
+class nodeStatsItem {
+    
+    nodeTag: string;
+    nodeUrl = ko.observable<string>();
+    
+    nodeType = ko.observable<clusterNodeType>();
+    nodeState = ko.observable<Raven.Client.ServerWide.RachisState>();
+
+    startTime = ko.observable<string>();
+    formattedUpTime = ko.observable<string>();
+    formattedStartTime: KnockoutComputed<string>;
+
+    disconnected = ko.observable<boolean>(true);
+    noData: boolean = true;
+    
+    even: boolean = false;
+
+    constructor(tag: string) {
+        this.nodeTag = tag;
+
+        this.formattedStartTime = ko.pureComputed(() => {
+            const start = this.startTime();
+            return start ? generalUtils.formatUtcDateAsLocal(start) : "";
+        });
+        
+        this.disconnected.subscribe((disconnected) => {
+            if (disconnected) {
+                this.noData = true;
+                this.nodeState(null);
+                this.nodeType(null);
+                this.nodeUrl(null);
+                this.startTime(null);
+                this.formattedUpTime(null);
+            }
+        })
+    }
+
+    update(data: Raven.Server.Dashboard.Cluster.Notifications.ClusterOverviewPayload) {
+        this.noData = false;
+        
+        this.nodeType(data.NodeType as clusterNodeType);
+        this.nodeState(data.NodeState);
+        this.nodeUrl(data.NodeUrl);
+        
+        this.startTime(data.StartTime);
+        this.formattedUpTime(this.getUpTime(data.StartTime));
+    }
+
+    private getUpTime(startTime: string) {
+        if (!startTime) {
+            this.formattedUpTime("a few seconds");
+        }
+        
+        return generalUtils.formatDurationByDate(moment(startTime));
+    }
+}
+
+class clusterOverviewWidget extends websocketBasedWidget<Raven.Server.Dashboard.Cluster.Notifications.ClusterOverviewPayload> {
+
+    view = require("views/resources/widgets/clusterOverviewWidget.html");
+    private clusterManager = clusterTopologyManager.default;
+    
+    private gridController = ko.observable<virtualGridController<nodeStatsItem>>();
+    private columnPreview = new columnPreviewPlugin<nodeStatsItem>();
+    
+    nodeStats = ko.observableArray<nodeStatsItem>([]);
+
+    constructor(controller: clusterDashboard) {
+        super(controller);
+
+        for (const node of this.controller.nodes()) {
+            const stats = new nodeStatsItem(node.tag());
+            this.nodeStats.push(stats);
+        }
+    }
+
+    getType(): Raven.Server.Dashboard.Cluster.ClusterDashboardNotificationType {
+        return "ClusterOverview";
+    }
+
+    compositionComplete() {
+        super.compositionComplete();
+
+        const grid = this.gridController();
+        grid.headerVisible(true);
+
+        this.gridController().customRowClassProvider(item => item.even ? ["even"] : []);
```

Comment:
do we need custom row class provider? I'm not sure.

Reply:
removed